src/octoprint/plugins/virtual_printer/__init__.py

```diff
@@ -232,16 +241,46 @@ def __init__(self, socket, *args, **kwargs):
                 super().__init__(*args, **kwargs)
 
             def _do_read(self, size):
+                data = b""
                 ready = select.select([self._socket], [], [], 2)
-                if self._socket not in ready[0]:
-                    raise socket.timeout()
-                return self._socket.recv(size)
+                if self._socket in ready[0]:
+                    data = self._socket.recv(size)
+
+                return data
 
             def _do_write(self, data):
                 ready = select.select([], [self._socket], [], 10)
-                if self._socket not in ready[1]:
-                    raise socket.timeout()
-                self._socket.sendall(data)
+                if self._socket in ready[1]:
+                    self._socket.sendall(data)
+
+        class PTYVirtualPrinterWrapper(VirtualPrinterWrapper):
```

Comment:
Is it really a dedicated `PTYVirtualPrinterWrapper` or is `FileVirtualPrinterWrapper` maybe the better term?

Reply:
I was just trying to keep things the same but File is definitely a better term. No arguments against changing it so I've done so.